Source/gg2/Scripts/GameServer/serviceJoiningPlayer.gml

```diff
@@ -78,6 +78,23 @@ case STATE_EXPECT_PASSWORD:
     break;
 
 case STATE_CLIENT_AUTHENTICATED:
+    //Check if server is full
+    var noOfPlayers;
+    noOfPlayers = ds_list_size(global.players);
+    if(global.dedicatedMode)
+        noOfPlayers -= 1;
+        
+    if(noOfPlayers >= global.playerLimit)
+    {
+        write_ubyte(socket, SERVER_FULL);
+        break;
+    }
+    
+    //Message lobby to update playercount if we became full

```

Comment:
This wouldn't really work there anymore. The importance of the new registration is that it shows the new player count after this player has joined, but now it's sent *before* the new player has actually joined.